### PR TITLE
make pipelines to support torch1.8.1 and torch1.9.0

### PR DESCRIPTION
**Pull request by liqunfu**
**Description**: make pipelines to support torch1.8.1 and torch1.9.0

**Motivation and Context**
- CppExtension is needed.
- CppExtension requires to be built with a matching torch version
- As a result, we need to build packages to supported torch versions

### CODE REVIEW

**Comment by snnn** [comment]
So I suppose they don't have an C API so you are directly using their C++ classes. Then, how about CentOS version and C++ ABI? Did they ever mention GCC's _GLIBCXX_USE_CXX11_ABI macro in their documents?

**Comment by liqunfu** [comment]
> So I suppose they don't have an C API so you are directly using their C++ classes. Then, how about CentOS version and C++ ABI? Did they ever mention GCC's _GLIBCXX_USE_CXX11_ABI macro in their documents?

Good point! CppExtension is the pytorch's standard way to build custom shared libraries at build time. However, its uses pytorch build process. It may not have the special handles we did with CentOS, etc. Changes with CppExtension is in https://github.com/microsoft/onnxruntime/pull/8027

**Review by asolms — APPROVED** [review]

**Review by thiagocrepaldi — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -402,7 +402,9 @@ def get_torch_version():
     version_number = version_number + local_version
 
 if wheel_name_suffix:
-    package_name = "{}-{}".format(package_name, wheel_name_suffix)
+    if not (enable_training and wheel_name_suffix == 'gpu'):
+        # for training packages, local version is used to indicate device types
+        package_name = "{}-{}".format(package_name, wheel_name_suffix)
```
Comment (snnn) [inline setup.py:407]: Thank you.

Comment (snnn) [reply setup.py:407]: Then the `wheel_name_suffix` is not used anywhere. Should the script exit if it was specified?

Comment (snnn) [reply setup.py:407]: You may also add such a test to https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/build.py#L1622

```python 
elif use_cuda and not ....
```

**Review by thiagocrepaldi — APPROVED** [review]

**Pull request merged**